MCprep_addon/materials/skin.py

```diff
@@ -703,6 +733,77 @@ def execute(self, context):
 		return {'FINISHED'}
 
 
+class MCPREP_OT_swap_skin_variant(bpy.types.Operator, spawn_util.VariationProp):
+	"""Apply the active UIlist skin to select characters"""
+	bl_idname = "mcprep.swap_skin"
+	bl_label = "Swap mob skin"
+	bl_description = "Swap the mobs variant"
+	bl_options = {'REGISTER', 'UNDO'}
+	
+	# def invoke(self, context, event):
+	# 	return context.window_manager.invoke_props_dialog(
+	# 		self, width=300 * util.ui_scale())
+
+	# def draw(self, context: Context):
+	# 	layout = self.layout
+	# 	self.draw_variation_ui(context, layout)
+	
+	@tracking.report_error
+	def execute(self, context: Context):
+		obj = context.object
+		if obj.type != 'ARMATURE':
+			self.report({'ERROR'}, "Please select an armature")
+			return {'CANCELLED'}
+		mats, skipped = getMatsFromSelected(obj, False)
+		mob_type = obj.get("MCPREP_mob_type", "Custom")
+		
+		if mob_type == "Custom":
+			self.report({'ERROR'}, "You are not allowed to use on Custom rig")
+			return {'CANCELLED'}
+		
+		texture_paths = self.get_texture_paths(mob_type)
+		name = mob_type.lower().replace(" ", "_")
+		if mob_type == "Villager":
+			self.doVillager(mats, texture_paths)
+		elif mob_type == "Zombie":
+			check_entity_texture(context)
+			if self.zombie_variation in ["ZOMBIE", "HUSK"]:
+				# Using Player model so convert player skin to 1.8 format
+				# loadSkinFile()
+				pass
+			else:
+				# Assign textures materials for drown
+				pass
+		elif mob_type == "Skeleton":
+			pass
+		
+		elif mob_type in ("Allay", "Vex"):
+			if mob_type == "Allay":
+				name += "/" + name
+			else:
+				name = "illager"
+			
+		else:
+			pass
+		return {'FINISHED'}
+
+	def doVillager(self, materials: List[Material], texture_paths: List[str]):
+		""" materials texture path order follows by profession, profession level, type """
+		prof_mat, biome_mat, level_mat = None, None, None
+		if mat[0].get("MCPREP_VILLAGER_PROFESSION"):
+				image = util.loadTexture(texture_paths[0])
+				proStat = generate.assert_textures_on_materials(image, [mat])
+			if mat.get("MCPREP_VILLAGER_BIOME"):
+				image = util.loadTexture(texture_paths[1])
+				biomeStat = generate.assert_textures_on_materials(image, [mat])
+			if mat.get("MCPREP_VILLAGER_LEVEL"):
+				image = util.loadTexture(texture_paths[2])
+				levelStat = generate.assert_textures_on_materials(image, [mat])
+			if not all([proStat, biomeStat, levelStat]):
+				self.report({'ERROR'}, "Something wrong happen during swap variant texture")
```

Comment:
Another nitpick on grammar, although it's pretty vague anyway. 
```suggestion
				self.report({'ERROR'}, "Something wrong happen while swapping the variant texture")
```